### ISSUE TITLE
Intel compiler support

### ISSUE THREAD

**Issue by termoshtt** [cmr/openblas-src#17]
I try to compile this crate with intel-compiler (ifort) and succeed by rewriting dylib specification in `build.rs`.
https://github.com/termoshtt/openblas-src/pull/1/files
Do you have any plan to support intel compiler?
If you interest, I will create PR to introduce `intel` feature to switch gfortran/ifort.

**Comment by IvanUkhov** [comment]
Thank you for the issue! Yes, it certainly would be great to support different configurations.

I suppose OpenBLAS detected by itself that `gfortran` was not available and switched to `ifort`, right? If it’s the case, the way to proceed is to fetch the compiled utilized from OpenBLAS’s log files and then link to the right libraries in the build script. I was looking it to this some time ago and was able to pull this information from OpenBLAS, but then I dropped that code as I had only one alternative, `gfortran`, and was not able to test what would happen with other potential compilers.

**Comment by IvanUkhov** [comment]
Have a look at the code that I’m removing in [this](https://github.com/cmr/openblas-src/commit/bd3af5d0dd80307744604a6cd31cf7719242ba94) commit.

**Comment by termoshtt** [comment]
Thanks link!

> the way to proceed is to fetch the compiled utilized from OpenBLAS’s log files or something and then link to the right libraries in the build script.

It seems to be more complicated than I intended, but it is feasible as you did.
My concern is that it is hard to test on CI using ifort since there is no option to use Intel compiler on CI as I sought. If I create gfortran/ifort switch, it will be tested only on my environment.
Does this crate permit this?

**Comment by IvanUkhov** [comment]
Yes, I think it’s worth it. There will be at least one confirmation that it’s working as intended.

**Comment by termoshtt** [comment]
Ok, I will create it. Thanks!

**Comment by IvanUkhov** [comment]
@termoshtt, is this still relevant?

**Comment by termoshtt** [comment]
> is this still relevant?

Thanks for notification.  [The above link](https://github.com/termoshtt/openblas-src/pull/1/files) is out-of-date completely. But Intel compiler support is still a valid issue.

- Since the OpenBLAS build system must support icc/ifort, we can use it through openblas-build.
- A remaining problem is how to test it on CI.
  - I expect we can use the oneAPI container https://hub.docker.com/r/intel/oneapi-basekit for this purpose, though we may need some struggle to setup it
